src/lib/proposer/proposer.ml

```diff
@@ -178,7 +180,20 @@ module Make (Inputs : Inputs_intf) :
               ( Ledger_builder_diff.With_valid_signatures_and_proofs.payments
                   diff
                 :> Payment.t list )
-            ~ledger:(Ledger_builder.ledger ledger_builder)
+            ~snarked_ledger_hash:(fun () ->
+              let open Or_error.Let_syntax in
+              match ledger_proof_opt with
+              | Some proof ->
+                  let snarked_ledger_hash =
+                    Ledger_proof.(statement proof |> statement_target) 
+                  in
+                  let%map snarked_ledger =
+                    Ledger_builder.snarked_ledger ledger_builder
+                      ~snarked_ledger_hash
+                  in
+                  Frozen_ledger_hash.of_ledger_hash
+                    (Ledger.merkle_root snarked_ledger)
```

Comment:
Question: if all we need is the hash here, do we even need to do the last two steps? Wouldn't the `snarked_ledger_hash` be equal to the `merkle_root snarked_ledger`?